examples/evaluator/golang/simple/evaluate/evaluator.go

```diff
@@ -11,14 +11,46 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
 package evaluate
 
 import (
+	"open-match.dev/open-match/examples"
 	harness "open-match.dev/open-match/pkg/harness/evaluator/golang"
 	"open-match.dev/open-match/pkg/pb"
 )
 
 // Evaluate is where your custom evaluation logic lives.
+// This sample evaluator sorts and deduplicates the input matches.
 func Evaluate(p *harness.EvaluatorParams) ([]*pb.Match, error) {
-	return p.Matches, nil
+	scoreInDescendingOrder := func(a, b *pb.Match) bool {
+		return a.GetProperties().GetFields()[examples.MatchScore].GetNumberValue() > b.GetProperties().GetFields()[examples.MatchScore].GetNumberValue()
+	}
+	by(scoreInDescendingOrder).Sort(p.Matches)
+
+	results := []*pb.Match{}
+
+	dedup := map[string]bool{}
+	for _, match := range p.Matches {
+		valid := true
+		ids := []string{}
+		for i := 0; valid && i < len(match.GetTicket()); i++ {
```

Comment:
nit feedback on reorganizing code here: 

for _, match := range p.Matches {
    if isNonCollidingMatch(match, selectedTickets) {
        // Add tickets to dedup
        // Add to match to results
   }
}
func isNonCollidingMatch(...) {
    for _, ticket := range match.GetTicket() {
    // If ticket id belongs to the selected players set, return false
   }
}